### ISSUE TITLE
Investigate how database connections are acquired during GraphQL query execution

### ISSUE THREAD

**Issue by joshhdawes** [UserOfficeProject/user-office-project-issue-tracker#1271]
### What is the problem and why is it a problem

While working on optimising some of the GraphQL queries used by UOP, further investigations using Apollo Studio have shown us that some optimised queries are still taking far longer than they should do in certain cases. After looking at several executions of the same query, it seems like excessively high execution times are the exception, and don't seem to depend on query parameters.

Unfortunately, we don't currently have enough logging/monitoring in place to dig any deeper - all we can tell is that some queries sometimes take much longer than they should do, with no indication of what part of the query execution process is actually taking up all the time.

Since databases query are a central part of executing GraphQL queries, and they themselves consist of 1) acquiring a connection from UOP's connection pool, and 2) executing the query using that connection, it seems sensible to investigate these two processes.

**Comment by janosbabik** [comment]
Hi @joshhdawes , do you have a list of the slow queries? It would be helpful to test them using EXPLAIN ANALYZE.

**Comment by joshhdawes** [comment]
@janosbabik @TCMeldrum would you be able to give Janos a list of the queries that you've optimised? Not sure how easy it'll be to apply here since there's a mixture of DB queries and API calls.

**Comment by joshhdawes** [comment]
@TCMeldrum did you get around to sending this stuff to ESS? Though to be honest I'm not sure if there's much point anymore.

**Comment by TCMeldrum** [comment]
Sorry this one slipped through the net on my end. Here is a list of the ones I can think of:
| Old Query                     | New Query                                    |
|-------------------------------|----------------------------------------------|
| getFaps                       | getFapsMinimal                               |
| getInstruments                | getInstrumentsMinimal                        |
| getCalls                      | getCallsMinimal                              |
| getCallByQuestionId           | getCallByAnswerId                            |
| getCall                       | getCallSubmissionDetails/getCallCloseDetails |
| getProposal                   | getProposalToClone                           |
| getCallsByInstrumentScientist | getCallsByInstrumentScientist                |
| answerTopic                   | answerTopic                                  |
| getGenericTemplates  | getGenericTemplatesForCopy  |

**Comment by joshhdawes** [comment]
@janosbabik did you still want to do anything with this issue?

**Comment by janosbabik** [comment]
No, I do not. However if you are still experiencing performance issues, my recommendation is to identify which SQL query is slow, analyse it (EXPLAIN ANALYZE) and modify it if needed. To determine which query is slow we could use the **log_min_duration_statement** configuration in postgresql.

**Comment by joshhdawes** [comment]
Thanks Janos. I'll close the issue as not-planned for now then.
